applications/upgradeability-by-proxy.md

```diff
@@ -0,0 +1,141 @@
+
+^ wat
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+Smart contract upgradeability is important, because bugs, and allowing for new
+features. Upgrades may be constrained, like needing to have votes to upgrade,
+or something. Very flexible, this proposal.
+
+It would be nice if it were easy to use this when building with the smart
+contracts pallet.
+
+Our team is interested in this because something.
+
+
+### Project Details
+
+Upgradeability can be implemented in user-land, or as part of the smart
+contracts pallet.
+
+We are opting to start this off by implementing it in user land to be able to
+iterate with more flexibility.
+
+When we're happy with the result, we'll see what can be migrated to ink as a
+directive to reduce boilerplate.
+
+We're going to start by implementing the proxy pattern. We need to figure out
+how to handle state (which will be stored in the proxy, probably?) migrations
+when the implementation is upgraded.
+
+We also need to figure out how to handle upgrade authorization. We want this to
+be customizable, so that each contract can have its own logic for permitting
+upgrades. Can the authorization code itself be upgradeable? I don't see why
+not. Should it be a third contract that is upgradeable by itself?
+
+There is the delegator pattern. We're going to start with a template that
+contains the migration/authorization code, but whose proxied methods need to be
+manually implemented by the smart contract developer. The idea is to focus on
+the mechanism itself and not on the syntactic sugar that ink should provide.
+Having this should allow folks to use it, even though it won't be super ergonomic.
+
+The second part will be attempting to extract this pattern into an ink directive,
+automatically generating the proxied methods. This brings us ergonomy of use, making
+upgradeability easy to opt in.
+
+Parts of this may be liftable to the smart contracts pallet. Out of scope
+though, for future plans.
+
+
+### Ecosystem Fit
+Are there any other projects similar to yours? If so, how is your project different?
+
+## Team :busts_in_silhouette:
+
+### Team members
+* Name of team leader
+* Names of team members	
+
+### Contact
+* **Contact Name:** Full name of the contact person (e.g. John Brown)
+* **Contact Email:** Contact email (e.g. john@duo.com)
+* Website
+
+### Legal Structure 
+* **Registered Address:** Address of your registered legal entity, if available. Please keep it on one line. (e.g. High Street 1, London LK1 234, UK)
+* **Registered Legal Entity:** Name of your registered legal entity, if available. (e.g. Duo Ltd.)
+
+### Team's experience
+
+- link to randomness pseudo-paper?
+- smart contract development?
+- icos?
+- IXO stuff?
+- https://forever.lifeonmars.pt/?
+
+### Team Code Repos
+* https://github.com/<your_repo_1>
+* https://github.com/<your_repo_2>
+
+### Team LinkedIn Profiles
+* https://www.linkedin.com/<person_1>
+* https://www.linkedin.com/<person_2>
```

Comment:
* https://www.linkedin.com/in/hugopeixoto/
* https://www.linkedin.com/in/jvsantos